Use getEntityKey instead of buildEntityKey

vitoravelino opened the pull request:
Signed-off-by: Vítor Avelino <vavelino@suse.com>

Fixes #3755

KlapTrap commented:
There are some `WARNING in Circular dependency detected:` when building the app.

vitoravelino commented:
> There are some `WARNING in Circular dependency detected:` when building the app.

I reverted 4 of the 14 changed files back to what it was before and added a note as we discussed.
